### ISSUE TITLE
[Item 20] 래퍼클래스와 인터페이스

### ISSUE THREAD

**Issue by gmelon** [2023-java-study/book-study#75]
p.132  에서 `래퍼 클래스 관용구와 함께 사용하면 인터페이스는 기능을 향상시키는 안전하고 강력한 수단이 된다` 는 문장은

Item 18에서 예로 들었던 `ForwardingSet` 처럼 인터페이스를 상속받으면서 컴포지션으로 해당 구현 클래스를 받아 안전하게 기능을 확장하는 경우를 말하는걸까요?

**Comment by ssstopeun** [comment]
네 저도 그렇게 이해했습니다. 기능을 추가할때 아이템 21에서 나오는 것처럼 인터페이스를 수정하는 것은 위험하고 인터페이스를 구현한 클래스를 수정하는 것도 안전하다고 할 수 없는데 인터페이스와 래퍼 클래스 관용구를 함께 사용한다면 기능을 확장할 때 기존 코드 수정없이 기능을 확장할 수 있기 때문에 안전하고 강력한 수단이 된다고 한 것 같습니다.
